lib/public/utilities/fetch/RemoteDataSource.js

```diff
@@ -10,60 +10,72 @@
  *  granted to it by virtue of its status as an Intergovernmental Organization
  *  or submit itself to any jurisdiction.
  */
-import { Observable, RemoteData } from '/js/src/index.js';
+import { RemoteData } from '/js/src/index.js';
 import { getRemoteData } from './getRemoteData.js';
-import { ObservableData } from '../ObservableData.js';
 
 /**
  * Service class providing observable data fetching process
  */
-export class RemoteDataSource extends Observable {
+export class RemoteDataSource {
     /**
      * Constructor
      */
     constructor() {
-        super();
-
         /**
-         * @type {ObservableData<RemoteData>}
+         * @type {ObservableData<RemoteData> | null}
          * @private
          */
-        this._observableData = new ObservableData(RemoteData.notAsked());
+        this._observableData = null;
         this._abortController = null;
     }
 
+    /**
+     * Sets the observable data to which source's data should be pushed to
+     *
+     * @param {ObservableData} observableData the observable data to which data should be pushed
+     * @return {void}
+     */
+    pipe(observableData) {
+        this._observableData = observableData;
+    }
+
     /**
      * Fetch the given endpoint to fill current data
      *
      * @param {string} endpoint the endpoint to fetch
      * @return {Promise<void>} resolves once the data fetching has ended
      */
     async fetch(endpoint) {
-        this._observableData.current = RemoteData.loading();
+        if (!this._observableData) {
+            return null;
+        }
```

Comment:
I think it't better not to throw here, because this way it allows clients to sometimes unplug observable and not change anything of the code, continuing to call fetch.
I don't see any advantage of throwing here?

Reply:
well, matter of preferences, e.g. if you throw then you know if you forgot to configure DataSource instance, but as I said up to you